schema/pgsql/schema.sql

```diff
@@ -258,25 +320,38 @@ CREATE TABLE rule (
     name citext NOT NULL,
     timeperiod_id bigint REFERENCES timeperiod(id),
     object_filter text,
-    is_active boolenum NOT NULL DEFAULT 'y',
+
+    changed_at bigint NOT NULL DEFAULT EXTRACT(EPOCH FROM NOW()) * 1000,
```

Comment:
I'm afraid this isn't portable to MySQL (like https://github.com/Icinga/icinga-notifications/pull/203#discussion_r1618987098).

Reply:
Precisely speaking, we have to provide \*nix ts by ourselves, just like https://github.com/Icinga/icinga-notifications/pull/215/files#diff-e9b5c34a7eeb261757ec19f5cd8fdd7ade8b32d33cbaf742262f967a261da59eR353

Reply:
What is the problem here? From a quick test, this seems to work fine except that the snippet I copy-pasted from Icinga DB to get a millisecond timestamp (`UNIX_TIMESTAMP() * 1000`) only seems to produce second precision, but that should hopefully be fixable with a different expression.

```
MariaDB [jbrost]> create table default_test (t bigint default (UNIX_TIMESTAMP() * 1000));
Query OK, 0 rows affected (0.015 sec)

MariaDB [jbrost]> insert into default_test () values ();
Query OK, 1 row affected (0.010 sec)

MariaDB [jbrost]> insert into default_test () values ();
Query OK, 1 row affected (0.003 sec)

MariaDB [jbrost]> insert into default_test () values ();
Query OK, 1 row affected (0.009 sec)

MariaDB [jbrost]> select * from default_test;
+---------------+
| t             |
+---------------+
| 1720689757000 |
| 1720689760000 |
| 1720689765000 |
+---------------+
3 rows in set (0.001 sec)
```

Reply:
Sorry, forgot to mention that OLD versions, like Maria 10.1, are the problem:

```
Server version: 10.1.48-MariaDB-1~bionic mariadb.org binary distribution

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [idb]> create table default_test (t bigint default (UNIX_TIMESTAMP() * 1000));
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '(UNIX_TIMESTAMP() * 1000))' at line 1
```

Reply:
FWIW on AL 2 the MariaDB version is even lower.

Reply:
> FWIW on AL 2 the MariaDB version is even lower.

I wouldn't use that as an argument, as this OS ships mariadb-5.5, which is something we aren't ever going to support. The default mariadb version on SLES 12.5 is `10.2`, so if such defaults work with that version, I would just drop support for `10.1`.

Reply:
```
Server version: 10.2.44-MariaDB-1:10.2.44+maria~bionic mariadb.org binary distribution

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [idb]> create table default_test (t bigint default (UNIX_TIMESTAMP() * 1000));
Query OK, 0 rows affected (0,021 sec)
```

(RHEL 8 has v10.3 btw.)

Reply:
@Al2Klimov Which of the versions currently mentioned in the documentation (MySQL (≥5.5), MariaDB (≥10.1)) would no longer be supported if we added that default? So is Maria DB 10.1 the only thing we'd lose?

https://github.com/Icinga/icinga-notifications/blob/f369a116868e8189f321c8ef579b190a3647636a/doc/02-Installation.md#L24

Reply:
We'd also lose MySQL 5.7, I guess:

```
Server version: 5.7.44 MySQL Community Server (GPL)

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MySQL [idb]> create table default_test (t bigint default (UNIX_TIMESTAMP() * 1000));
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(UNIX_TIMESTAMP() * 1000))' at line 1
```

```
Server version: 8.0.38 MySQL Community Server - GPL

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MySQL [idb]> create table default_test (t bigint default (UNIX_TIMESTAMP() * 1000));
Query OK, 0 rows affected (0,067 sec)
```

Reply:
I just checked the available MariaDB version for all of our build targets:

- Debian
    - 11: 10.5.23
    - 12: 10.11.6
- Ubuntu
    - 20.04: 10.3.39
    - 22.04: 10.6.18
    - 23.04: 10.11.2
    - 23.10: 10.11.8
    - 24.04: 10.11.8
- Amazon Linux
    - 2: 5.5.68
    - 2023: 10.5
- Fedora
    - 37: 10.5.23
    - 38: 10.5.23
    - 39: 10.5.25
    - 40: 10.11.8
- openSUSE
    - 15.4: 10.6.15
    - 15.5: 10.6.18
    - 15.6: 10.11.8
- RHEL (versions from Rocky because reasons)
    - 8: 10.11.6
    - 9: 10.11.6
- SLES:
    - 12.5: 10.2.44
    - 15.3: 10.5.8
    - 15.4: 10.6.18
    - 15.5: 10.6.18
    - 15.6: 10.11.8

@Al2Klimov: You mentioned MariaDB 10.1 as a lower bound. Thus, only Amazon Linux 2 with 5.5.68 should be an issue.

Reply:
We changed the schema and have dropped the default. @sukhwinder33445 has already implemented it on the :spider_web: side.

Reply:
https://github.com/Icinga/icinga-notifications/pull/191#issuecomment-2222885560